crates/matrix-sdk-sqlite/src/event_cache_store.rs

```diff
@@ -507,11 +507,22 @@ impl EventCacheStore for SqliteEventCacheStore {
                             "INSERT INTO events(chunk_id, room_id, event_id, content, position) VALUES (?, ?, ?, ?, ?)"
                         )?;
 
-                        for (i, event) in items.into_iter().enumerate() {
+                        let items = items
+                            .into_iter()
+                            .filter_map(|event| {
+                                let Some(event_id) = event.event_id() else {
+                                    error!(%room_id, "Trying to push an event with no ID");
+                                    return None;
+                                };
+
+                                Some((event_id.to_string(), event))
```

Comment:
I would pull this closure out into a separate variable so  this line reads like:

```rust
.filter_map(invalid_events)
```

Not a strong preference but a suggestion.

Reply:
Done!